modules/data-entry/src/main/frontend/src/questionnaire/NumberQuestion.jsx

```diff
@@ -195,17 +195,17 @@ function NumberQuestion(props) {
     { height: Math.max(100, sliderMarks.length*30) + "px" } : undefined
 
   // Callback function for our min/max
-  let hasMinMaxValueError = (text) => {
+  let getMinMaxValueError = (text) => {
     let value = 0;
     if (typeof(text) === "undefined" || text === "") {
       // The custom input has been unset
-      return false;
+      return null;
     }
 
     if (dataType === "long") {
       // Test that it is an integer
       if (!/^[-+]?\d*$/.test(text)) {
-        return true;
+        return "The values must be an integer";
```

Comment:
Should we switch plurality here based on whether it's a range or not? Also, since we're tweaking these messages anyway, we might want to use the term "whole number" instead of the technical "integer".

```suggestion
        return isRange ? "The values must be whole numbers" : "The value must be a whole number";
```